localstack/services/opensearch/cluster.py

```diff
@@ -248,19 +311,78 @@ def __init__(self, enabled: bool, endpoint: str) -> None:
             self.url = None
 
 
+class EndpointProxy:
+    def __init__(self, base_url: str, forward_url: str, custom_endpoint: str) -> None:
+        super().__init__()
+        self.base_url = base_url
+        self.forward_url = forward_url
+        self.custom_endpoint = custom_endpoint
+        self.routing_rules = None
+
+    def register(self):
+        _url = urlparse(self.base_url)
+        self.routing_rules = register_cluster(
+            host=_url.hostname,
+            path=_url.path,
+            forward_url=self.forward_url,
+            custom_endpoint=self.custom_endpoint,
+        )
+
+    def unregister(self):
+        try:
+            for rule in self.routing_rules:
+                from localstack.services.edge import ROUTER
+
+                ROUTER.remove_rule(rule)
+        except KeyError:
+            pass
```

Comment:
When can this `KeyError` happen? Is this really something that's not noteworthy at all? An `except pass` without any logging or anything else really needs to be a conscious decision on the control-flow. And if so, please add a comment explaining _why_ it is okay to just pass on these raised errors.

Reply:
I had them here and there, sometimes it happened that routes were being removed from the router that were already gone. I will add some logging (I had some, then saw that the existing method only passed, should have stuck to my code). Also it is probably better to replace this with a more generic except.